src/App.js

```diff
@@ -78,15 +81,44 @@ const routes = [
   },
 ];
 
-const Base = props => (
-  <div className={css.App}>
-    <Nav {...props} />
-    {props.children}
-  </div>
-);
+class Base extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      registerModal: false,
+      signInModal: false,
+    };
+  }
+  handleModalToggle = modalName => {
+    const modal =
+      modalName === 'registerModal' ? 'signInModal' : 'registerModal';
+    this.setState({ [modalName]: !this.state[modalName], [modal]: false });
```

Comment:
```suggestion
    this.setState((prevState) => {[modalName]: !prevState[modalName], [otherModal]: false});
```
`prevState` should be used to reference the previous state

Reply:
yo, nice i didnt know we could do that